clippy_lints/src/escape.rs

```diff
@@ -105,15 +103,15 @@ fn is_argument(map: &hir::map::Map<'_>, id: HirId) -> bool {
 }
 
 impl<'a, 'tcx> Delegate<'tcx> for EscapeDelegate<'a, 'tcx> {
-    fn consume(&mut self, cmt: &cmt_<'tcx>, mode: ConsumeMode) {
-        if let Categorization::Local(lid) = cmt.cat {
+    fn consume(&mut self, cmt: &Place<'tcx>, mode: ConsumeMode) {
+        if let PlaceBase::Local(lid) = cmt.base {
```

Comment:
In this file there should also be a check that `cmt.projections` is empty.

Reply:
I just added it to every method of the EUV. It didn't break anything, so I think this should be ok?